ragnarok_formats/src/model.rs

```diff
@@ -57,21 +57,74 @@ where
 
 #[derive(Debug, ByteConvertable)]
 #[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
-pub struct PositionKeyframeData {
+pub struct ScaleKeyframeData {
     pub frame: u32,
-    pub position: Point3<f32>,
+    pub scale: Vector3<f32>,
+    reserved: f32,
 }
-
 #[derive(Clone, Debug, ByteConvertable)]
 #[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
 pub struct RotationKeyframeData {
     pub frame: u32,
     pub quaternions: Quaternion<f32>,
 }
 
+#[derive(Debug, ByteConvertable)]
+#[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
+pub struct TranslationKeyframeData {
+    pub frame: u32,
+    pub translate: Vector3<f32>,
+    reserved: f32,
+}
+
+#[derive(Debug, ByteConvertable)]
+#[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
+pub struct TexturesKeyframeData {
+    pub texture_index: u32,
+    #[new_derive]
+    pub texture_keyframe_count: u32,
+    #[repeating(texture_keyframe_count)]
+    pub texture_keyframes: Vec<TextureKeyframeData>,
+}
+
+// List of texture operation types - https://rathena.org/board/topic/127587-rsm2-file-format/
+// 0: Texture translation on the X axis. The texture is tiled.
+// 1: Texture translation on the Y axis. The texture is tiled.
+// 2: Texture multiplication on the X axis. The texture is tiled.
+// 3: Texture multiplication on the Y axis. The texture is tiled
+// 4: Texture rotation around (0, 0). The texture is not tiled.
+#[derive(Debug, ByteConvertable)]
+#[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
+pub enum TextureOperation {
+    TranslationX,
+    TranslationY,
+    ScaleX,
+    ScaleY,
+    Rotation,
+}
```

Comment:
For documentation style comments like this you should really use Rusts doc comments (`///`). That will make these comments show up in peoples IDEs as well as documentation generated by `cargo doc`.

You can attach doc comments to almost anything, for example the varians of an enum. So I would instead write the coments like this:

```rust
/// List of texture operation types.
/// See: https://rathena.org/board/topic/127587-rsm2-file-format/
#[derive(Debug, ByteConvertable)]
#[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
pub enum TextureOperation {
    /// Texture translation on the X axis. The texture is tiled.
    TranslationX,
    /// Texture translation on the Y axis. The texture is tiled.
    TranslationY,
    /// Texture multiplication on the X axis. The texture is tiled.
    ScaleX,
    /// Texture multiplication on the Y axis. The texture is tiled
    ScaleY,
    /// Texture rotation around (0, 0). The texture is not tiled.
    Rotation,
}
```